File: 1-100/35/35.py
# rotation generator
def genrot(str):
	orig = str[:]
	while 1:
		str = str[-1:] + str[:-1]
		if str == orig: break
		yield str
	
# Generate primes up to 1,000,000
print 'Sieving...'
limit = 1000000
arr = [True] * limit
def sieve(x):
	global arr,limit
	for i in xrange(x*2,limit,x): arr[i] = False
map(sieve, xrange(2,limit**1/2))
primes = [i for i in xrange(2,limit) if arr[i]]

# Slow test for primeness
def isprime(x):
	global primes
	# Quick test - any even or '5' digits?
	for c in str(x):
		if ['1','3','7','9'].count(c) == 0: return False;
	# Longer, more thorough test: is it in the primes array?
	try:
		primes.index(x)
		return True
	except:
		return False

def isCirc(x):
	for y in genrot(str(x)):
		if not isprime(int(y)):
			return False
	return True

# Test for circularness
print 'Circulizing...'
circs = [x for x in primes if isCirc(x)]

File: 1-100/35/35.cpp
int primes[1000000];
int isprime[1000000];

main() {
  int i, j, k, m, nprimes=0, s, p, ok, count=0;
  char test[13];
  for(i=2;i&lt;1000000;i++) {
    s = sqrt(i);
    p=1;
    for(j=0;j&lt;nprimes && primes[j] &lt;= s;j++)
      if(i%primes[j] == 0) {
        p=0;
        break;
      }
    if(isprime[i] = p)
      primes[nprimes++] = i;
  }
  for(i=0;i&lt;nprimes;i++) {
    sprintf(test, "%d%d", primes[i], primes[i]);
    k = strlen(test)/2;
    ok=1;
    for(j=k-1;j&gt;=1;j--) {
      test[j+k] = '\0';
      sscanf(test+j, "%d", &m);
      if(!isprime[m]) {
        ok=0;
        break;
      }
    }
    if(ok)
      count++;
  }
  printf("%d\n", count);
}
